fe1-web/src/features/wallet/network/DigitalCashMessageApi.ts

```diff
@@ -0,0 +1,76 @@
+import { publish } from 'core/network';
+import { getCoinChannel, Hash, KeyPair, PopToken, PublicKey } from 'core/objects';
+
+import { Transaction } from '../objects/transaction';
+import { DigitalCashStore } from '../store';
+import { PostTransaction } from './messages';
+
+import getBalance = DigitalCashStore.getBalance;
+
+const makeErr = (err: string) => `Sending the transaction failed: ${err}`;
+
+/**
+ * Requests a digital cash transaction post
+ * @param from the popToken to send it with
+ * @param to the destination public key
+ * @param amount the value of the transaction
+ * @param laoId the id of the lao in which to send the transaction
+ */
+export function requestSendTransaction(
+  from: PopToken,
+  to: PublicKey,
+  amount: number,
+  laoId: Hash,
+): Promise<void> {
+  const transactionStates = DigitalCashStore.getTransactionsByPublicKey(
+    laoId.valueOf(),
+    from.publicKey.valueOf(),
+  );
+
+  if (transactionStates.length === 0) {
+    throw new Error(makeErr('no transaction out were found for this public key'));
```

Comment:
I noticed when testing manually that this exception isn't caught. (I get the metro screen) It would be better if it triggered a toast.

![image](https://user-images.githubusercontent.com/15657735/172365322-32c48813-1995-48db-bfc0-0a8594175c96.png)

Reply:
Thanks, that was supposed to be caught !